plugins/repository-hdfs/hadoop-client-api/build.gradle

```diff
@@ -1,16 +1,47 @@
-apply plugin: 'elasticsearch.build'
-apply plugin: 'com.gradleup.shadow'
+apply plugin: 'elasticsearch.java'
+
+sourceSets {
+  patcher
+}
+
+configurations {
+  thejar {
+    canBeResolved = true
+  }
+}
 
 dependencies {
-  implementation "org.apache.hadoop:hadoop-client-api:${project.parent.versions.hadoop}"
+  thejar("org.apache.hadoop:hadoop-client-api:${project.parent.versions.hadoop}") {
+    transitive = false
+  }
+
+  patcherImplementation 'org.ow2.asm:asm:9.7.1'
+  patcherImplementation 'org.ow2.asm:asm-tree:9.7.1'
 }
 
-tasks.named('shadowJar').configure {
-  exclude 'org/apache/hadoop/util/ShutdownHookManager$*.class'
+def outputDir = layout.buildDirectory.dir("patched-classes")
+
+def patchTask = tasks.register("patchClasses", JavaExec) {
+  inputs.files(configurations.thejar).withPathSensitivity(PathSensitivity.RELATIVE)
+  inputs.files(sourceSets.patcher.output).withPathSensitivity(PathSensitivity.RELATIVE)
+  outputs.dir(outputDir)
+  classpath = sourceSets.patcher.runtimeClasspath
+  mainClass = 'org.elasticsearch.hdfs.patch.HdfsClassPatcher'
+  doFirst {
+    args(configurations.thejar.singleFile, outputDir.get().asFile)
+  }
 }
 
-['jarHell', 'thirdPartyAudit', 'forbiddenApisMain', 'splitPackagesAudit'].each {
-  tasks.named(it).configure {
-    enabled = false
+tasks.named('jar').configure {
+  dependsOn(patchTask)
+  dependsOn(configurations.thejar)
+
+  from(outputDir)
```

Comment:
I think this can actually just become `from(patchTask)` which will automatically use the task's outputs and you can remove the `dependsOn` above.